### PR TITLE
Add pprof links to landing page

### PR DESCRIPTION
**Pull request by SuperQ**
Make it easier for users to fetch and download pprof data for debugging.
* Add a link to https://pprof.me for easy sharing.

image

### CODE REVIEW

**Review by metalmatze — APPROVED** [review]
That is an excellent idea!
It will make it a lot easier to get profiles from users.

**Comment by SuperQ** [comment]
@bboreham What do you think of adding something like this to Prometheus `/status`?

File: web/landing_page.html [hunk 1]
```diff
@@ -30,6 +30,14 @@ <h1>{{.Name}}</h1>
       </div>
       {{ end }}
       {{ .ExtraHTML }}
+      <div id="pprof">
+      Go pprof debugging profile data:
+      <ul>
+        <li><a href="debug/pprof/heap">heap usage (memory)</a>
+        <li><a href="debug/pprof/profile">CPU usage (takes some time to gather)</a>
+      </ul>
+      Try posting these to <a href="https://pprof.me" target="_blank">pprof.me</a>
```
Comment (bboreham) [inline web/landing_page.html:39]:
```suggestion
       "To share with an expert, you can upload to <a href="https://pprof.me" target="_blank">pprof.me</a>
```

File: web/landing_page.html [hunk 2]
```diff
@@ -30,6 +30,14 @@ <h1>{{.Name}}</h1>
       </div>
       {{ end }}
       {{ .ExtraHTML }}
+      <div id="pprof">
+      Go pprof debugging profile data:
```
Comment (bboreham) [inline web/landing_page.html:34]:
```suggestion
      To download a detailed report of resource usage (pprof format, from the Go runtime):
```

File: web/landing_page.html [hunk 3]
```diff
@@ -30,6 +30,14 @@ <h1>{{.Name}}</h1>
...
+      <ul>
+        <li><a href="debug/pprof/heap">heap usage (memory)</a>
+        <li><a href="debug/pprof/profile">CPU usage (takes some time to gather)</a>
```
Comment (bboreham) [inline web/landing_page.html:37]: You could add `?seconds=10` to make it less alarming for the newcomer.

**Review by bboreham — COMMENTED** [review]
Good idea; I had some suggestions about wording.

What would you think if I said it should offer both pprof.me and flamegraph.com ?

**Comment by bboreham** [comment]
> What do you think of adding something like this to Prometheus `/status`?

Sure.  I guess the page with most info about memory is `/tsdb-status`.  Suspect what people really want is a top-level breakdown like "40% on samples, 30% on scraping, 20% on remote-write".  Which we don't know how to do accurately, but might be able to approximate in a useful way.

File: web/landing_page.html (hunk 1 above)
Comment (SuperQ) [reply web/landing_page.html]: Hmm, I don't like the `expert` term. I made a minor tweak to the suggestion.

File: web/landing_page.html (hunk 3 above)
Comment (SuperQ) [reply web/landing_page.html:37]: Yes, but that has a high chance of missing a scrape interval. I was actually thinking about making this `60s` so we gather a few scrapes. Or capture instances where people reduce the scrape interval.

**Comment by dswarbrick** [comment]
It might be worth mentioning somewhere in docs that these links will be ineffectual (and most likely just bounce back to the landing page, albeit with additional path suffixes each time) unless the exporter imports `net/http/pprof`.

**Comment by SuperQ** [comment]
My plan is to add pprof handling to the exporter-toolkit.

**Pull request merged**